src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -14882,4 +14887,35 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
 
 #endif // defined(DEBUG) || defined(LATE_DISASM)
 
+//----------------------------------------------------------------------------------------
+// IsOppositeOfPrevMov:
+//    Check if previous instruction was a mov where dst was current src and src was current dst.
+//
+// Arguments:
+//    ins - The current instruction
+//    dst - The current destination
+//    src - The current source
+//
+// Return Value:
+//    true if previous instruction moved from current dst to src.
+
+bool emitter::IsOppositeOfPrevMov(instruction ins, regNumber dst, regNumber src)
+{
+    assert(ins == INS_mov);
+
+    if (emitLastIns != nullptr && emitLastIns->idIns() == INS_mov)
+    {
+        // Check if we did same move in prev instruction except dst/src were switched.
+        regNumber prevDst = emitLastIns->idReg1();
+        regNumber prevSrc = emitLastIns->idReg2();
+
+        if ((prevDst == src) && (prevSrc == dst))
```

Comment:
Might as well check for and suppress identical movs too.

Reply:
You mean when `src == dst`? If yes, we already do it [here](https://github.com/dotnet/runtime/blob/e1297d6a18c8be2c0948b4f486541c06649ee4eb/src/coreclr/src/jit/emitarm64.cpp#L4016).

Reply:
Or I think you mean this:
```asm
mov x1, x2
mov x1, x2  # <-- omit this
```